contracts/staking/stakeManager/StakeManagerExtension.sol

```diff
@@ -128,13 +128,18 @@ contract StakeManagerExtension is StakeManagerStorage, Initializable, StakeManag
     function migrateValidatorsData(uint256 validatorIdFrom, uint256 validatorIdTo) external {       
         for (uint256 i = validatorIdFrom; i < validatorIdTo; ++i) {
             ValidatorShare contractAddress = ValidatorShare(validators[i].contractAddress);
-            if (contractAddress != ValidatorShare(0)) {
-                validators[i].delegatorsReward = contractAddress.validatorRewards_deprecated().add(INITIALIZED_AMOUNT);
+            if (contractAddress != ValidatorShare(0) && !validators[i].migrated) {
+                validators[i].migrated = true;
+                // move validator rewards out from ValidatorShare contract
+                validators[i].rewards = contractAddress.validatorRewards_deprecated().add(INITIALIZED_AMOUNT);
                 validators[i].delegatedAmount = contractAddress.activeAmount();
                 validators[i].commissionRate = contractAddress.commissionRate_deprecated();
+                validators[i].delegatorsReward = INITIALIZED_AMOUNT;
```

Comment:
Do we need it here? We are already doing it after this if-else.

Reply:
No need, probably forgot to remove it from if clause